### PR TITLE
Improve the cache-fix plugin build

### PR DESCRIPTION
**Pull request by bigdaz**
- Enable caching
- Enable parallel test execution
- Use `maven-publish` instead of `Upload` task
- Use Java toolchains to compile with JDK8

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -67,48 +71,40 @@ sourceSets {
...
-    descriptorDestination = file("$buildDir/ivy/ivy.xml")
-    uploadDescriptor = true
-    configuration = configurations.runtime
 }
 
 tasks.withType(Test).configureEach {
-    dependsOn install
+    dependsOn publish
     systemProperty "local.repo", localRepo.toURI()
+    maxParallelForks = 4
```
Comment (ghale) [inline build.gradle:105]: Did you find that this actually improved anything?  When I've tried this before, with the android builds these tests execute, I found that it didn't improve the execution time at all, but it made my system nearly unusable while the tests were running.  It appeared that it was almost completely I/O bound, so adding more parallelism didn't buy us anything.

File: build.gradle [hunk 2]
```diff
@@ -67,48 +71,40 @@ sourceSets {
...
-
+gradlePlugin {
     plugins {
         androidCacheFixPlugin {
             id = 'org.gradle.android.cache-fix'
             displayName = 'Gradle Android cache fix plugin'
-            tags = ['android', 'cache', 'fix']
-            version = project.version
+            description = 'Gradle plugin to fix Android caching problems'
+            implementationClass = 'org.gradle.android.AndroidCacheFixPlugin'
```
Comment (ghale) [inline build.gradle:80]: I think this means we can remove `src/main/resources/META-INF/gradle-plugins/org.gradle.android-cache-fix.properties` now, right?

**Review by ghale — CHANGES_REQUESTED** [review]
Thanks for doing this.  Left a few minor comments, but otherwise it looks good.

File: build.gradle (hunk 1 above)
Comment (bigdaz) [reply build.gradle:105]: I can't say I really noticed, but tests seemed super-slow without it. 
Looking at the CI job, it's probably not any faster. I'll remove this.

File: build.gradle (hunk 2 above)
Comment (bigdaz) [reply build.gradle:80]: Good catch. Removed

**Comment by bigdaz** [comment]
@ghale I'll merge this as-is and improve the test in a second pass. I'll also look at having a single `taskGraph.whenReady` hook, although I'm not overly concerned about leaving it with a separate hook per Workaround.

**Comment by bigdaz** [comment]
@ghale Oops, that comment was meant for the other PR.

File: build.gradle (hunk 1 above)
Comment (runningcode) [reply build.gradle]: Haha, yeah I did this too at some point :)

**Pull request merged**